apps/channels/clients/connect_client.py

```diff
@@ -29,3 +45,53 @@ def create_channel(self, connect_id: UUID, channel_source: str) -> UUID:
         response = self.client.post(url, json={"connectid": str(connect_id), "channel_source": channel_source})
         response.raise_for_status()
         return UUID(response.json()["channel_id"])
+
+    def decrypt_messages(self, key: bytes, messages: list[Message]) -> list[str]:
+        """
+        Decrypts the `MessagePayload` list using the provided key and verifies the message authenticity
+
+        Raises:
+            ValueError if the message authenticity cannot be trusted
+        """
+        decrypted_messages = []
+        for message in messages:
+            nonce = base64.b64decode(message["nonce"])
+            tag = base64.b64decode(message["tag"])
+            ciphertext = base64.b64decode(message["ciphertext"])
+            cipher = AES.new(key, AES.MODE_GCM, nonce=nonce)
+            decrypted_messages.append(cipher.decrypt_and_verify(ciphertext, tag).decode("utf-8"))
```

Comment:
There is an `encrypt_message` method below. It is nice to have symmetrical methods for this kind of thing. Maybe consider extracting this to a `decrypt_message(key, ciphertext, tag, nonce)` method.

You could also move the b64 encoding there to make it completely symmetrical: `assert message == decrypt_message(key, *encrypt_message(key, message))`

Reply:
I like this. Like this? https://github.com/dimagi/open-chat-studio/pull/1041/commits/42ad7f26f758ea118ca1c36b37d54e62fd59175f